src/test/tests/databases/silo.py

```diff
@@ -635,4 +635,46 @@ def TurnOffSetsByName(silr, cat, names):
 DeleteAllPlots()
 CloseDatabase(data_path("silo_ghost_test_data/specmix_ucd.pdb"))
 
+#
+# Test the way silo reads in curve data with expressions
+#
+OpenDatabase(data_path("silo_pdb_test_data/multi_rect2d.silo"))
+DefineCurveExpression("log10", "log+10")
+DefineCurveExpression("waveSquared", "wave*wave")
+DefineCurveExpression("line2", "line+line")
+DefineCurveExpression("waveLine", "line+wave")
```

Comment:
Super! Thanks for adding tests too.

Can you add an expression to test which is just an aliasing expression, DefineCurveExpression("alias","wave")

Also, can you add similar tests to ``src/test/tests/databases/curve.py`` to test curve expressions there?

Reply:
Do you want me to move these tests from silo.py to curve.py, or do you want me to make new tests in cuve.py? If the latter, do you want me to use data from "curve_test_data" or continue using silo_pdb?

Reply:
Keep the tests you've written here. They test a case where curves are being read from a database that has multi-block objects. 

Ideally, we'd loop over these tests twice, once using a database with multi-block objects and once using a database without any multi-block objects. But, I went looking for *silo* data files with curves in them that are single block...I couldn't find any. Maybe you can -- or maybe you don't mind updating some test code there to produce curves in a Silo file that has only single block (for example, modify ``src/dev/data/datagen/testall.C`` with a call to Silo's ``DBPutCurve()`` function to write some curve data to the databases ``testall.C`` produces, update the ``silo_hdf5_test_data.7z`` and ``silo_pdb_test_data.7z`` test tarballs. I thought that was a little too onerous to request. So, the easy thing is to not seek a *loop over two Silo database solutions* here and instead, test these curve expressions from databases with multi-block objects in ``silo.py`` and test these same (or similar) curve expressions from a different, single block, database that has curves...best example(s) being in ``curve_test_data``. Does that make sense?

Also, it occurs to me, of the 4 expressions you define and test for here, I don't think any one is there is different from any other. They all do a summation or multiplication. So, I don't think 3 of the 4 tests are actually needed. I mean, are you aware of any cases where ``line+line`` works but ``wave*wave`` doesn't? So, far, the only two *unique* failures I think we were seeing was the *alias* case which gave a flat (all zeros) result and a case involving some arithmetic like ``wave*wave``. So, I honestly think those two cases are the only ones that require testing...again that is assuming you haven't seen additional *unique* failures. No sense in adding tests that don't actually provide any new information.

Reply:
Yes, that makes sense. I didn't know that curve_test_data housed silo files because I was expecting the file extensions to be .silo. Will work on getting these tests up.

I see what you're saying about the extra tests. I did not see any failures that depended on what type of arithmetic was being done. I'll remove the extra tests

Reply:
``curve_test_data`` does not house silo files. That is why I was asking that you essentially perform the same tests in ``curve.py`` which is where we test the ``Curve2D`` database plugin.

Reply:
Okay, I think I understand what you're asking for. I'll push my changes when ready for you to take another look at.

Reply:
Okay, I just pushed my changes to curve.py. Let me know if this is what you wanted.